eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/utils/charsequence/TrailingLinesTrimmed.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.utils.charsequence;
+
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+
+
+/**
+ * A {@link CharSequence} decorator that trims trailing lines.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class TrailingLinesTrimmed extends AbstractCachingCharSequence
+{
+    public TrailingLinesTrimmed(final CharSequence text)
+    {
+        super(new Factory<CharSequence>()
+        {
+            @Override
+            public CharSequence create()
+            {
+                return removeTrailingLines(text);
+            }
+        });
+    }
+
+
+    // http://stackoverflow.com/a/25058538
+    // (Note: A Character.isWhitespace() might not be good for html, it messed up the html format before.)
+    private static CharSequence removeTrailingLines(CharSequence text)
+    {
+        if (text.length() == 0)
+        {
+            return text;
+        }
+
+        while (text.charAt(text.length() - 1) == '\n')
```

Comment:
This won't work if there is also spaces on these empty lines, also it's a very inefficient way of doing it.